src/app/util/ember-compatibility-functions.cpp

```diff
@@ -272,6 +273,23 @@ CHIP_ERROR ReadSingleClusterData(FabricIndex aAccessingFabricIndex, const Concre
         return SendFailureStatus(aPath, attributeReport, Protocols::InteractionModel::Status::UnsupportedAttribute, nullptr);
     }
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
+        Access::RequestPath requestPath{ .cluster = aPath.mClusterId, .endpoint = aPath.mEndpointId };
+        Access::Privilege requestPrivilege = Access::Privilege::kView; // TODO: get actual request privilege
+        CHIP_ERROR err                     = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
+        if (err != CHIP_NO_ERROR)
+        {
+            AttributeReportIB::Builder attributeReport = aAttributeReports.CreateAttributeReport();
+            ReturnErrorOnFailure(aAttributeReports.GetError());
+            TLV::TLVWriter backup;
+            attributeReport.Checkpoint(backup);
+            auto status = (err == CHIP_ERROR_ACCESS_DENIED) ? Protocols::InteractionModel::Status::UnsupportedAccess
+                                                            : Protocols::InteractionModel::Status::Failure;
+            return SendFailureStatus(aPath, attributeReport, status, &backup);
```

Comment:
This is wrong for a wildcard read.  For a wildcard read, the UnsupportedAccess case just means the path is skipped, without error.  That means that this check needs to be done at a point where we know whether our path is coming from a wildcard or not, which is up the callstack from here.  Or we need to pass that information down here.  This is the most important change needed here.

Reply:
Yeah I was worrying less about spec order in this PR and more about getting checks in so they can be hooked up subsequently and final order can be hammered out. It's already a struggle to get the code to match spec order for the reasons you mentioned. I was hoping some of that could be deferred, but maybe we have to bite the bullet sooner.

Reply:
I put the check in the correct spot for wildcard read so it works (tested in chip-repl).